src/main/java/de/tum/in/www1/artemis/service/iris/session/IrisChatSessionService.java

```diff
@@ -128,10 +132,19 @@ public void requestAndHandleResponse(IrisSession session) {
         parameters.put("exercise", exercise);
         parameters.put("course", exercise.getCourseViaExerciseGroupOrCourseMember());
         parameters.put("latestSubmission", "");
+        parameters.put("buildResult", "");
         var participations = studentParticipationRepository.findByExerciseIdAndStudentIdWithEagerResultsAndLegalSubmissions(exercise.getId(), chatSession.getUser().getId());
         if (participations.size() >= 1) {
             var latestSubmission = participations.get(0).getSubmissions().stream().max(Comparator.comparing(Submission::getSubmissionDate));
-            latestSubmission.ifPresent(submission -> parameters.put("latestSubmission", submission));
+            if (latestSubmission.isPresent()) {
+                var submission = latestSubmission.get();
+                parameters.put("latestSubmission", submission);
+                if (submission.getLatestResult() != null && submission.getLatestResult().getId() != null) {
+                    var programmingSubmission = programmingSubmissionRepository.findByResultId(latestSubmission.get().getLatestResult().getId());
+                    programmingSubmission.ifPresent(result -> parameters.put("buildResult", result));
```

Comment:
Why do you use the id of the result to retrieve a programming submission only to the use the result of that submission again?
Can't you directly load the programming submission or even the wanted result?